src/components/dev-hub/star-rating.js

```diff
@@ -0,0 +1,180 @@
+import PropTypes from 'prop-types';
+import StarIcon from '~components/dev-hub/icons/star-icon';
+import styled from '@emotion/styled';
+import { size, fontSize, screenSize } from './theme';
+
+const StyledContainer = styled.div`
+    display: flex;
+    flex-direction: column;
+
+    @media ${screenSize.mediumAndUp} {
+        flex-direction: row;
+    }
+`;
+
+const StyledStar = styled(StarIcon)`
+    width: ${size.mediumLarge};
+    height: ${size.mediumLarge};
+    margin-right: ${size.mediumLarge};
+    box-sizing: content-box;
+
+    @media ${screenSize.mediumAndUp} {
+        padding: 4px;
+        width: auto;
+        height: auto;
+        margin-right: ${({ isActive }) => (isActive ? '0' : size.xsmall)};
+
+        &:hover {
+            cursor: ${({ isClicked }) => (isClicked ? 'default' : 'pointer')};
+        }
+    }
+`;
+
+const StyledStarsContainer = styled.div`
+    display: flex;
+    align-items: center;
+
+    @media ${screenSize.mediumAndUp} {
+        margin-left: ${size.large};
+    }
+`;
+
+const StyledTitle = styled.span`
+    font-family: 'Fira Mono', monospace;
+    font-weight: bold;
+    font-size: ${fontSize.small};
+`;
+
+const initialState = {
+    first: false,
+    second: false,
+    third: false,
+    fourth: false,
+    fifth: false,
+    isClicked: false,
+};
+
+const reducer = (state, action) => {
+    let nextState = { ...initialState };
+    switch (action) {
+        case 'fifth':
+            nextState = { ...nextState, fifth: true };
+        case 'fourth':
+            nextState = { ...nextState, fourth: true };
+        case 'third':
+            nextState = { ...nextState, third: true };
+        case 'second':
+            nextState = { ...nextState, second: true };
+        case 'first':
+            nextState = { ...nextState, first: true };
+            break;
+        case 'clicked':
+            nextState = { ...state, isClicked: true };
+            break;
+        case 'clear':
+            nextState = { ...initialState };
+            break;
+        default:
+            throw new Error();
+    }
+
+    return nextState;
+};
+
+const StarRating = ({ onFirstsClick, onMiddleClick, onLatestClick }) => {
+    const [state, dispatch] = useReducer(reducer, initialState);
+
+    const onClickHandler = useCallback(
+        handler => {
+            if (!state.isClicked) {
+                handler && handler();
+                dispatch('clicked');
+            }
+        },
+        [state.isClicked]
+    );
+
+    const onHoverHandler = useCallback(
+        star => {
+            !state.isClicked && dispatch(star);
+        },
+        [state.isClicked]
+    );
+    const onLeaveHandler = useCallback(() => {
+        !state.isClicked && dispatch('clear');
+    }, [state.isClicked]);
+
+    return (
+        <StyledContainer>
+            <StyledTitle>Rate this article</StyledTitle>
+            <StyledStarsContainer onMouseLeave={onLeaveHandler}>
+                <StyledStar
```

Comment:
Would is be possible to use a `button` tag as this is semantically a button? We have the `Button` component which should be easy to style without making it `primary` or `secondary`

That also allows us to get CSS like the `cursor` built-in already.